src/Routes/RosPage/RosPage.js

```diff
@@ -21,14 +24,53 @@ class RosPage extends React.Component {
 
     constructor(props) {
         super(props);
+
+        this.state = {
+            page: 1,
+            perPage: 5
+        };
+
+        this.onSetPage = this.onSetPage.bind(this);
+        this.onPerPageSelect = this.onPerPageSelect.bind(this);
     }
 
     async componentDidMount() {
         await window.insights.chrome.auth.getUser();
-        this.props.fetchSystems();
+        this.fetchWithParams();
+    }
+
+    onSetPage(event, page) {
+        const { perPage } = this.state;
+        const pagination = { page, perPage };
+        this.updatePagination(pagination);
+    }
+
+    onPerPageSelect(event, perPage) {
+        const page = 1;
+        const pagination = { page, perPage };
+        this.updatePagination(pagination);
+    }
```

Comment:
You don't really need these 2 functions, you can call directly `updatePagination`

When setting page: 
```JSX
onSetPage={ (_e, page) => this.updatePagination({ page, perPage: this.state.perPage })}
```
When setting perPage:
```JSX
onPerPageSelect={(_e, perPage) => this.updatePagination({ page: 1, perPage })}
```